src/profile-logic/merge-compare.js

```diff
@@ -132,21 +132,42 @@ export function mergeProfilesForDiffing(
       );
     }
     const profile = profiles[i];
-    let thread = profile.threads[selectedThreadIndex];
+    let thread = { ...profile.threads[selectedThreadIndex] };
     transformStacks[i] = profileSpecific.transforms[selectedThreadIndex];
     implementationFilters.push(profileSpecific.implementation);
 
     // We adjust the categories using the maps computed above.
     // TODO issue #2151: Also adjust subcategories.
-    thread.stackTable.category = adjustCategories(
-      thread.stackTable.category,
-      translationMapsForCategories[i]
-    );
-    thread.frameTable.category = adjustNullableCategories(
-      thread.frameTable.category,
-      translationMapsForCategories[i]
+    thread.stackTable = {
+      ...thread.stackTable,
+      category: adjustCategories(
+        thread.stackTable.category,
+        translationMapsForCategories[i]
+      ),
+    };
+    thread.frameTable = {
+      ...thread.frameTable,
+      category: adjustNullableCategories(
+        thread.frameTable.category,
+        translationMapsForCategories[i]
+      ),
+    };
+
```

Comment:
These are part of another PR, which is under review. I kept it here to figure out if merging is easier like this.

Reply:
The code has been merged into main. Dismissing the comments here ^